flake.nix

```
@@ -1,19 +1,47 @@
 {
   description = "Deku development environment";
-  inputs.nixpkgs.url = "github:nixos/nixpkgs";
-  inputs.flake-utils.url = "github:numtide/flake-utils";
-  inputs.esy-fhs.url = "github:d4hines/esy-fhs";
-  inputs.esy-fhs.inputs.nixpkgs.follows = "nixpkgs";
-  outputs = { self, nixpkgs, flake-utils, esy-fhs }:
+
+  # Setup trusted binary caches
+  nixConfig = {
+    trusted-substituters = [
+      "https://cache.nixos.org/"
+      "https://deku.cachix.org"
+    ];
+  };
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    # Use this branch until we can merge it in the overlays
+    ocaml-overlays.url = "github:anmonteiro/nix-overlays/ulrikstrid/ocaml-trunk";
+    ocaml-overlays.inputs.nixpkgs.follows = "nixpkgs";
+
+    nix-npm-buildpackage.url = "github:serokell/nix-npm-buildpackage";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, nix-npm-buildpackage, ocaml-overlays }:
     flake-utils.lib.eachDefaultSystem (system:
-      let pkgs = import nixpkgs { inherit system; }; in
+      let        
+        pkgs = ocaml-overlays.legacyPackages."${system}";
```

What about something like this?
```suggestion
       pkgs = import nixpkgs {
           inherit system;
           overlays = [ocaml-overlays.overlay]
        };
```
This makes it more clear where to add other overlays as we need them.